platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/Threshold.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.utility;
+
+/**
+ * Describes various mathematically important thresholds.
+ */
+public enum Threshold {
+    /**
+     * In order to meet this threshold, the part must be &ge;1/3 of the whole.
+     */
+    STRONG_MINORITY,
+    /**
+     * In order to meet this threshold, the part must be &gt;1/2 of the whole.
+     */
+    MAJORITY,
+    /**
+     * In order to meet this threshold, the part must be &gt;2/3 of the whole.
+     */
+    SUPER_MAJORITY;
+
+    /**
+     * Check if a threshold is satisfied by a given ratio.
+     *
+     * @param part  the numerator
+     * @param whole the denominator
+     * @return true if the threshold is satisfied by the provided ratio
+     * @throws IllegalArgumentException if part is negative or whole is non-positive or zero, or if part is greater than
+     *                                  whole
+     */
+    public boolean isSatisfiedBy(final long part, final long whole) {
+        return switch (this) {
```

Comment:
Instead of this switch statement, an enum can have a lambda as a parameter. that way you just call the lambda